src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.cs

```diff
@@ -238,14 +258,27 @@ private void CloseWorker()
         /// longer in use. In particular, make sure to avoid letting a connection lease cross an <see langword="await"/>
         /// boundary, as it will prevent code in the asynchronous operation from using the existing connection.
         /// </remarks>
-        private PooledConnection GetPooledConnection()
-            => new PooledConnection(this, GetConnection());
+        private PooledConnection GetPooledConnection(out SqlConnection connection)
+        {
+            var result = new PooledConnection(this, GetConnection());
+            connection = result.Connection;
+            return result;
+        }
 
         public void Initialize(Solution solution)
         {
+            if (_shutdownTokenSource.IsCancellationRequested)
+            {
+                // Someone tried to get a connection *after* a call to Dispose the storage system
+                // happened.  That should never happen.  We only Dispose when the last ref to the
+                // storage system goes away.  Once that happens, it's an error for there to be any
+                // future or existing consumers of the storage service.  So nothing should be doing
+                // anything that wants to get an connection.
+                throw new InvalidOperationException();
```

Comment:
❔ Should this be `ObjectDisposedException`?

Reply:
good question.  i'm somewhat ok with either approach.